x-pack/plugins/infra/common/http_api/custom_dashboards_api.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as rt from 'io-ts';
+
+const AssetTypeRT = rt.type({
+  /**
+   * For now we have only one asset type, so it's a literal type.
+   * In the future it can be changed to a union type:
+   * `rt.union([rt.literal('host'), rt.literal('container'), rt.literal('pod')])`
+   */
+  assetType: rt.literal('host'),
+});
+
+const CustomDashboardRT = rt.intersection([
+  AssetTypeRT,
+  rt.type({
+    dashboardSavedObjectIdList: rt.array(rt.string),
```

Comment:
Same as comment for SO schema, even more so in this case if this endpoint is gonna be public.